Framework/LiveData/inc/MantidLiveData/ADARA/ADARAPackets.h

```diff
@@ -483,12 +494,8 @@ class MANTID_LIVEDATA_DLL BeamMonitorConfigPkt : public Packet {
       return (0);
   }
 
-  double distance(uint32_t index) const {
-    if (index < beamMonCount())
-      return *reinterpret_cast<const double *>(&m_fields[(index * 6) + 5]);
-    else
-      return (0.0);
-  }
+private:
+  double distance(uint32_t index) const { return extractDouble(index, 5); }
```

Comment:
`extractDouble()` checks `index < detBankSetCount()`, but previously this function was checking `index < beamMonCount()`. Also, the factor of 6 seems to have been lost. This code change doesn't look safe.